### PR TITLE
[DROOLS-3968] Move off JUnit 4 and onto JUnit Vintage

### PR DESCRIPTION
**Pull request by triceo**
Only merge after https://github.com/kiegroup/submarine-bom/pull/16.

### CODE REVIEW

File: drools/drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/ComplexRulesTest.java [hunk 1]
```diff
@@ -517,7 +517,7 @@ public void testCompareDate() {
     }
 
     @Test
-    public void testCompareDateWithString() {
+    public void testCompareDateWithString() { // FIXME fails on non-US locales
```
Comment (triceo) [inline drools/drools-model/drools-model-compiler/src/test/java/org/drools/modelcompiler/ComplexRulesTest.java]: This is not related to my PR. I just found out that this particular test fails when I use cs_CZ locale on Linux.

**Comment by triceo** [comment]
Failure appears unrelated to my changes. More importantly, it wasn't there last week, when I was first preparing this PR.

**Comment by baldimir** [comment]
"[ERROR] 'dependencies.dependency.version' for junit:junit:jar is missing." Please provide similar PR also for submarine-cloud. Otherwise the PR pipeline will be broken.

**Comment by triceo** [comment]
@baldimir I forgot about the cloud repo. Fixed now.

**Pull request merged**